bee-protocol/src/worker/propagator.rs

```diff
@@ -62,27 +56,27 @@ async fn propagate<B: StorageBackend>(
             // parent in question turns out to be a SEP.
 
             // Determine OTRSI/YTRSI of parent1
-            let (parent1_otrsi, parent1_ytrsi) = match tangle.get_solid_entry_point_index(&parent1).await {
-                Some(parent1_sepi) => (IndexId::new(parent1_sepi, parent1), IndexId::new(parent1_sepi, parent1)),
-                None => match tangle.get_metadata(&parent1).await {
-                    Some(parent1_md) => (parent1_md.otrsi().unwrap(), parent1_md.ytrsi().unwrap()),
-                    None => continue,
-                },
-            };
-
-            // Determine OTRSI/YTRSI of parent2
-            let (parent2_otrsi, parent2_ytrsi) = match tangle.get_solid_entry_point_index(&parent2).await {
-                Some(parent2_sepi) => (IndexId::new(parent2_sepi, parent2), IndexId::new(parent2_sepi, parent2)),
-                None => match tangle.get_metadata(&parent2).await {
-                    Some(parent2_md) => (parent2_md.otrsi().unwrap(), parent2_md.ytrsi().unwrap()),
-                    None => continue,
-                },
-            };
+
+            let mut parent_otrsis = Vec::new();
+            let mut parent_ytrsis = Vec::new();
+
+            for parent in parents.iter() {
+                let (parent_otrsi, parent_ytrsi) = match tangle.get_solid_entry_point_index(parent).await {
+                    Some(parent_sepi) => (IndexId::new(parent_sepi, *parent), IndexId::new(parent_sepi, *parent)),
+                    None => match tangle.get_metadata(parent).await {
+                        Some(parent_md) => (parent_md.otrsi().unwrap(), parent_md.ytrsi().unwrap()),
+                        None => continue,
```

Comment:
this `continue` now refers to it's parent for-loop but it used to refer to the while-let. Just use loop labels here to fix this (I love Rust for supporting that)